src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Serializer/ProtobufOtlpCommonFieldNumberConstants.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.Serializer;
+
+using System.Diagnostics.CodeAnalysis;
+
+/// <summary>
+/// Defines field number constants for fields defined in
+/// <see href="https://github.com/open-telemetry/opentelemetry-proto/blob/v1.2.0/opentelemetry/proto/common/v1/common.proto"/>.
+/// </summary>
+[SuppressMessage("StyleCop.CSharp.NamingRules", "SA1310:Field names should not contain underscore", Justification = "These names describe the nested names and properties in the .Proto file.")]
```

Comment:
This message is across 4 files, should be suppress this message for `Serializer` folder name?

Reply:
Just FYI `.editorconfig` supports heirarchy so you should be able to drop a `.editorconfig` file into the folder and turn off this one rule for all the files in here. May need to add root/top-level to the main one if it isn't already there. Some examples on contrib if you want to take a look.

Reply:
I wouldn't want to suppress this for the entire directory because that would affect the Serializer classes.

I'll take a look at what can be accomplished with the `.editorconfig`, maybe I can filter that exception to only "*Constants.cs" files.

Reply:
Done. I got this working with the editorconfig and verified that it doesn't apply to the Serializer classes :)